src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -500,44 +503,54 @@ public static void close(Closeable closeable) {
      */
     public static URI getUriByFilename(String filename) throws CheckstyleException {
         // figure out if this is a File or a URL
-        URI uri;
-        try {
-            final URL url = new URL(filename);
-            uri = url.toURI();
-        }
-        catch (final URISyntaxException | MalformedURLException ignored) {
-            uri = null;
+        URI uri = null;
+        if (filename.startsWith(CLASSPATH_URL_PREFIX)) {
+            final String path = filename.substring(CLASSPATH_URL_PREFIX.length());
+            uri = toUri(path);
         }
-
-        if (uri == null) {
-            final File file = new File(filename);
-            if (file.exists()) {
-                uri = file.toURI();
-            }
-            else {
-                // check to see if the file is in the classpath
+        else {
+            if (filename.charAt(0) != '/') {
                 try {
-                    final URL configUrl;
-                    if (filename.charAt(0) == '/') {
-                        configUrl = CommonUtil.class.getResource(filename);
-                    }
-                    else {
-                        configUrl = ClassLoader.getSystemResource(filename);
-                    }
-                    if (configUrl == null) {
-                        throw new CheckstyleException(UNABLE_TO_FIND_EXCEPTION_PREFIX + filename);
-                    }
-                    uri = configUrl.toURI();
+                    final URL url = new URL(filename);
+                    uri = url.toURI();
+                }
+                catch (final URISyntaxException | MalformedURLException ignored) {
+                    // ignored
```

Comment:
No good, we need to rethrow all outside of Checkstyle

Reply:
it's history code. we should support windows path, ie: 'C:/xxx'

Or we change  precheck logic just on http ?
```java
            if (filename.startsWith("http://") || filename.startsWith("https://")) {
                try {
                    final URL url = new URL(filename);
                    uri = url.toURI();
                }
                catch (final URISyntaxException | MalformedURLException ignored) {
                    throw new CheckstyleException(UNABLE_TO_FIND_EXCEPTION_PREFIX + filename, ignored);
                }
            }
```

Reply:
@qxo Don't mark thinks as resolved when you have a question back or the item isn't resolved from the initial review. Just say done when you believe you have resolved it and allow the reviewers to mark things resolved to confirm.

Reply:
ok:)